### PR TITLE
Feature : add condition

### PR DESCRIPTION
**Pull request by RomulusED69**
see [ISSUE-9](https://github.com/gordalina/GordalinaMixpanelBundle/issues/9)

### CODE REVIEW

**Comment by tristanbes** [comment]
The use case is the following:

On your Symfony application, you can have some actions for forms that do 2 things:
1. display your form
2. save your form

With this PR you would be able to do something like this:

```php
   /**
     * @Mixpanel\Track("Display contact form", condition="request.getMethod() in ['GET']")
     * @Mixpanel\Track("Contact form submission", condition="request.getMethod() in ['POST']")
     */
    public function yourAction()
    {
        // ...
    }
```

If the condition evaluates to false, the event won't be sent to Mixpanel

File: README.md [hunk 1]
```diff
@@ -216,6 +216,21 @@ e.g: `@Mixpanel\Expression("<expression>")` or `@Mixpanel\Set("<property>", valu
 **Note**: All `id` properties can be omitted, as they will be set with the id of
 the current user in `security.context`
 
+### Condition
+
+In all annotations, you can add condition with Expression
```
Comment (tristanbes) [inline README.md]:
```suggestion
In all annotations, you can add conditions with [Expression Language](https://symfony.com/doc/current/components/expression_language.html)
```

File: README.md [hunk 2]
```diff
@@ -216,6 +216,21 @@ e.g: `@Mixpanel\Expression("<expression>")` or `@Mixpanel\Set("<property>", valu
 **Note**: All `id` properties can be omitted, as they will be set with the id of
 the current user in `security.context`
 
+### Condition
```
Comment (tristanbes) [inline README.md]:
```suggestion
### Send an event based on a condition
```

File: Annotation/Annotation.php [hunk 3]
```diff
@@ -24,4 +24,9 @@ abstract class Annotation
      * @var string
      */
     public $project;
+
+    /**
+     * @var Gordalina\MixpanelBundle\Annotation\Expression
```
Comment (tristanbes) [inline Annotation/Annotation.php]: you should remove this type so you can simplify your `condition` usage like this : https://symfony.com/doc/4.1/routing/conditions.html

(`condition="method...`)

File: Annotation/Annotation.php [hunk 4]
```diff
@@ -24,4 +24,9 @@ abstract class Annotation
...
+     */
+    public $condition = true;
```
Comment (tristanbes) [inline Annotation/Annotation.php]:
```suggestion
    public $condition;
```

File: Mixpanel/EventListener/ControllerListener.php [hunk 5]
```diff
@@ -99,7 +101,7 @@ private function updateDataAnnotation(Annotation\Annotation $annotation, Request
         }
 
         if ($value instanceof Annotation\Expression) {
-            $element = $this->expressionLanguage->evaluate($value->expression, $request->attributes->all());
+            $element = $this->expressionLanguage->evaluate($value->expression, array_merge($request->attributes->all(), ['request' => $request]));
```
Comment (tristanbes) [inline Mixpanel/EventListener/ControllerListener.php]: You should take example on how Symfony used expression language for its routing there: 
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Routing/Matcher/Dumper/CompiledUrlMatcherDumper.php#L428-L433

**Comment by tristanbes** [comment]
Hello @gordalina. 

This PR is ready, the code has been live tested on our project. 👍

**Comment by gordalina** [comment]
Awesome, thank you

**Pull request merged**